tests/hardware/plugins/temperature/test_tc4820.py

```diff
@@ -66,11 +66,17 @@ def _get_message(len: int) -> tuple[int, bytes, Any]:
     )
 
 
-_TERM_CHARS = set("*^")
-"""Some random ASCII characters, plus the correct start and terminator chars."""
+def _get_term_chars() -> list[str]:
+    chars = set("*^")
+
+    for c in range(0, 128, 10):
+        chars.add(chr(c))
+
+    return sorted(chars)
 
-for c in range(0, 128, 10):
-    _TERM_CHARS.add(chr(c))
+
+_TERM_CHARS = _get_term_chars()
```

Comment:
Not sure why this complicated. Why not just:
```python
_TERM_CHARS = ["^", "*"] + [chr(c) for c in range(0, 128, 10)]
```
And it is not random, but rather deterministic, always the same characters.

Reply:
It's so that we don't end up with duplicates for the first two. It doesn't matter much, but we don't really want to run the exact same test twice.

Reply:
But you are. There's no randomness in here. Regardless of using the function or the list, you get exactly the same collection of characters, right?

Reply:
We won't get duplicates now because I use a set, then convert it to a list.

I'm not sure whether the `range(0, 128, 10)` bit includes `^` and `*`, so I add them separately. I could check, but then if we change the range (e.g. to include more chars) then we might accidentally remove them and we definitely should test those two, because they have a special meaning.

I suppose another way of doing it would be to just include every char between 0 and 127 inclusive (i.e. the whole ASCII range). That's not so many tests, really.

Reply:
I just realised I describe it as "random ASCII characters" but I meant they were arbitrary, not actually random :stuck_out_tongue:. I should probably change this comment